### PR TITLE
fix(jwt): strong type for SigningKeyCallback error

### PR DESCRIPTION
**Pull request by hkleungai**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
    - `SigningKeyCallback` should only accept `Error` type or `null` type `error` argument.
    - See [here](https://github.com/auth0/node-jsonwebtoken/blob/74d5719bd03993fcf71e3b176621f133eb6138c0/verify.js#L95-L97) for the original source code.

### CODE REVIEW

File: types/jsonwebtoken/index.d.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@
 //                 Ivan Sieder <https://github.com/ivansieder>
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
 //                 Nandor Kraszlan <https://github.com/nandi95>
+//                 Jimmy Leung <https://github.com/hkleungai>
```
Comment (jakebailey) [inline types/jsonwebtoken/index.d.ts:15]: Are you looking to become an owner/reviewer of this package? If you aren't, don't need to edit this block: https://github.com/DefinitelyTyped/DefinitelyTyped#definition-owners

File: types/jsonwebtoken/index.d.ts [hunk 2]
```diff
@@ -51,83 +52,75 @@ export interface SignOptions {
      * - ES512:    ECDSA using P-521 curve and SHA-512 hash algorithm
      * - none:     No digital signature or MAC value included
      */
-    algorithm?: Algorithm | undefined;
-    keyid?: string | undefined;
+    algorithm?: Algorithm;
```
Comment (jakebailey) [inline types/jsonwebtoken/index.d.ts:55]: Please don't remove all of the `| undefined`; it's unrelated to the change your PR states to fix, and will break users of `exactOptionalPropertyTypes`.


See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

File: types/jsonwebtoken/jsonwebtoken-tests.ts [hunk 3]
```diff
@@ -4,8 +4,8 @@
  * Created by using code samples from https://github.com/auth0/node-jsonwebtoken.
  */
 
-import jwt = require("jsonwebtoken");
-import fs = require("fs");
+import jwt = require('jsonwebtoken');
```
Comment (jakebailey) [inline types/jsonwebtoken/jsonwebtoken-tests.ts:7]: Your PR contains a load of style changes; would you consider only fixing the type issue that you have? I know that our repo contains a prettier config, but at the moment, we don't enforce it and I can't find the actual change in this PR with all of this other unrelated stuff.

**Review by jakebailey — DISMISSED** [review]

File: types/jsonwebtoken/index.d.ts (hunk 1 above)
Comment (hkleungai) [reply types/jsonwebtoken/index.d.ts:15]: Issue addressed.

File: types/jsonwebtoken/jsonwebtoken-tests.ts (hunk 3 above)
Comment (hkleungai) [reply types/jsonwebtoken/jsonwebtoken-tests.ts]: Issue addressed.

File: types/jsonwebtoken/index.d.ts (hunk 2 above)
Comment (hkleungai) [reply types/jsonwebtoken/index.d.ts]: Issue addressed.

**Review by johnnyreilly — APPROVED** [review]

**Comment by hkleungai** [comment]
Ready to merge

**Pull request merged**